packages/core/react/use-chat.ts

```diff
@@ -62,6 +63,11 @@ export function useChat({
   const hookId = useId()
   const chatId = id || hookId
 
+  //Useful to execute logic when the chat is mounted.
+  if (onStart) {
```

Comment:
This doesn't run on mount but during and on every render.

Reply:
At the moment, this is implemented as....

```
onStart()
const { ... } = useChat()
```

If you need to run something on mount, then just add an effect at the call site?

Reply:
@jaredpalmer Yes, what you said makes sense. 
I was actually trying to find a way to avoid running  `const { data, mutate } = useSWR<Message[]>([api, chatId], null, {`. 
if certain situation happens so we can avoid running it.

Can you think in a good use case for that?

Reply:
Didn't want to said "on mount" it's more on hook initialization.